### PR TITLE
Rewrite the `Condition` documentation for clarity and flow.

### PR DESCRIPTION
**Pull request by sergetron**
Rewrites the `Condition` documentation for clarity and flow.

### CODE REVIEW

File: docs/conditions.md [hunk 1]
```diff
@@ -6,50 +6,54 @@ weight: 11
...
 
-## Syntax
+A `Condition` resource in Tekton allows you to conditionalize the execution of `Tasks` within a `Pipeline`.
+You define each `Condition` within your `PipelineRun` definition and then conditionalize each desired `Task`
+in the corresponding `Pipeline` definition. 
 
-To define a configuration file for a `Condition` resource, you can specify the
-following fields:
+The `Condition` resource runs its own container image that executes the logic that evaluates your chosen condition.
+This container runs to completion and must return an exit code value of `0` fo the `check` to be successful; all
```
Comment (pritidesai) [inline docs/conditions.md:23]: typo `fo` == `from`

File: docs/conditions.md [hunk 2]
```diff
@@ -6,50 +6,54 @@ weight: 11
...
-  - [Check](#check)
-  - [Parameters](#parameters)
-  - [Resources](#resources)
-- [Labels and Annotations](#labels-and-annotations)
-- [Examples](#examples)
+## Overview
 
-## Syntax
+A `Condition` resource in Tekton allows you to conditionalize the execution of `Tasks` within a `Pipeline`.
+You define each `Condition` within your `PipelineRun` definition and then conditionalize each desired `Task`
```
Comment (pritidesai) [inline docs/conditions.md:19]: Would be great to include link to `PipelineRun` doc here i.e. 

[PipelineRun](pipelineruns.md)

File: docs/conditions.md [hunk 3]
```diff
@@ -6,50 +6,54 @@ weight: 11
...
+
+A `Condition` definition supports the following fields:
 
 - Required:
   - [`apiVersion`][kubernetes-overview] - Specifies the API version, for example
-    `tekton.dev/v1alpha1`.
-  - [`kind`][kubernetes-overview] - Specify the `Condition` resource object.
-  - [`metadata`][kubernetes-overview] - Specifies data to uniquely identify the
-    `Condition` resource object, for example a `name`.
+    `tekton.dev/v1beta1`.
```
Comment (pritidesai) [inline docs/conditions.md:35]: Conditions are still in `v1alpha1` 😉

File: docs/conditions.md [hunk 4]
```diff
@@ -6,50 +6,54 @@ weight: 11
...
+  - [`kind`][kubernetes-overview] - Identifies this resource object as a `Condition` object.
```
Comment (pritidesai) [inline docs/conditions.md:36]: I think older description was clear too _Specify the `Condition` resource object._ i.e. the kind has to set to `Condition`

File: docs/conditions.md [hunk 5]
```diff
@@ -6,50 +6,54 @@ weight: 11
...
-### Check
-
-The `check` field is required. You define a single check to define the body of a `Condition`. The 
-check must specify a [`Step`](./tasks.md#steps). The container image runs till completion. The container 
-must exit successfully i.e. with an exit code 0 for the condition evaluation to be successful. All other 
-exit codes are considered to be a condition check failure.
+### Specifying the condition `check` 
 
-### Description
+The `check` field (required) specifies a single piece of evaluation logic that you want to run before the
```
Comment (pritidesai) [inline docs/conditions.md:49]: how about having `required` in bold?

File: docs/conditions.md (hunk 1 above)
Comment (sergetron) [reply docs/conditions.md:23]: Thanks for catching this. That's actually supposed to be "for". Fixed!

File: docs/conditions.md (hunk 2 above)
Comment (sergetron) [reply docs/conditions.md:19]: Added link.

File: docs/conditions.md (hunk 4 above)
Comment (sergetron) [reply docs/conditions.md:36]: This is how we have it in all other docs, so I think I'll leave it. `kind` is already linked and self-explanatory.

File: docs/conditions.md (hunk 5 above)
Comment (sergetron) [reply docs/conditions.md:49]: I think bold would negate the parentheses - this is a piece of additional or "by-the-way" information while we explain what the field does.

File: docs/conditions.md (hunk 3 above)
Comment (sergetron) [reply docs/conditions.md:35]: Oops, didn't know that - fixed!

File: docs/conditions.md [hunk 6]
```diff
@@ -6,50 +6,54 @@ weight: 11
...
-## Syntax
+A `Condition` resource in Tekton allows you to conditionalize the execution of `Tasks` within a `Pipeline`.
+You define each `Condition` within your [`PipelineRun` definition](pipelineruns.md) and then conditionalize
+each desired [`Task`](tasks.md) in the corresponding `Pipeline` definition. 
 
-To define a configuration file for a `Condition` resource, you can specify the
-following fields:
+The `Condition` resource runs its own container image that executes the logic that evaluates your chosen condition.
+This container runs to completion and must return an exit code value of `0` for the `check` to be successful; all
+other exit codes result in a failure.
```
Comment (sbwsg) [inline docs/conditions.md:24]: Wdyt about qualifying the meaning of "failure" here?  If I remember right the pipeline doesn't end up failed, but the conditioned task in the pipeline is skipped.

Comment (jerop) [reply docs/conditions.md:24]: the branch of the conditioned task is skipped as well

Comment (jerop) [reply docs/conditions.md:24]: the branch is made up of the dependent tasks -- based on ordering (defined using runAfter) and resource dependencies (e.g. results)

Comment (jerop) [reply docs/conditions.md:24]: the branch of the conditioned task is skipped as well -- the branch is made up of the dependent tasks (based on ordering dependencies defined using runAfter and resource dependencies e.g. results)

Comment (sergetron) [reply docs/conditions.md:24]: All great points - reworded to qualify the outcome in the event of an unsuccessful condition check.

**Pull request merged**